api/queries_pr.go

```diff
@@ -205,9 +205,9 @@ func (pr *PullRequest) ChecksStatus() (summary PullRequestChecksStatus) {
 	return
 }
 
-func (c Client) PullRequestDiff(baseRepo ghrepo.Interface, prNum int) (string, error) {
+func (c Client) PullRequestDiff(baseRepo ghrepo.Interface, pr *PullRequest) (string, error) {
 	url := fmt.Sprintf("https://api.github.com/repos/%s/pulls/%d",
-		ghrepo.FullName(baseRepo), prNum)
+		ghrepo.FullName(baseRepo), pr.Number)
```

Comment:
Was there a strong reason to change the call signature of this method? The way I see it, accepting an int instead of a PullRequest makes it more versatile, and if the caller already has a PullRequest reference they can easily pass the `pr.Number` as argument themeselves

Reply:
@vilmibm and I talked about this at one point. I think we thought it was more flexible to pass in the PR, but now that you mention that it sounds like it is LESS flexible now. I'll revert the change.